### ISSUE TITLE
`no-deprecated-modulo-syntax` rule

### ISSUE THREAD

**Issue by kazupon** [intlify/eslint-plugin-vue-i18n#495]
**Please describe what the rule should do:**

vue-i18n v10 will drop modulo Interpolation (rails-style interpolation).
docs is here:
https://vue-i18n.intlify.dev/guide/essentials/syntax.html#rails-i18n-format

I would like to provide this rule to support migration, and support with `--fix` in this rule too.

**What category should the rule belong to?**

- [x] Enforces code style (layout)
- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

✗ BAD :
```json
{
  "hello": "%{msg} world"
}
```

✓ GOOD:
```json
{
  "hello": "{msg} world"
}
```

**Additional context**
Not really relevant to this eslint plugin, but I will output a warning with message-compiler if it is also using modulo.
